includes/class-post-types.php

```diff
@@ -119,61 +119,19 @@ public static function register_inbox_post_type() {
 		\register_post_meta(
 			Inbox::POST_TYPE,
 			'_activitypub_object_id',
-			array(
-				'type'              => 'string',
-				'single'            => true,
-				'description'       => 'The ID (ActivityPub URI) of the object that the inbox item is about.',
-				'sanitize_callback' => 'sanitize_url',
-			)
+			self::get_object_id_meta_args()
 		);
 
 		\register_post_meta(
 			Inbox::POST_TYPE,
 			'_activitypub_activity_type',
-			array(
-				'type'              => 'string',
-				'description'       => 'The type of the activity',
-				'single'            => true,
-				'show_in_rest'      => true,
-				'sanitize_callback' => function ( $value ) {
-					$value  = ucfirst( strtolower( $value ) );
-					$schema = array(
-						'type'    => 'string',
-						'enum'    => Activity::TYPES,
-						'default' => 'Create',
-					);
-
-					if ( \is_wp_error( \rest_validate_enum( $value, $schema, '' ) ) ) {
-						return $schema['default'];
-					}
-
-					return $value;
-				},
-			)
+			self::get_activity_type_meta_args( 'Create' )
 		);
 
 		\register_post_meta(
 			Inbox::POST_TYPE,
 			'_activitypub_activity_actor',
-			array(
-				'type'              => 'string',
-				'single'            => true,
-				'description'       => 'The type of the local actor that received the activity.',
-				'show_in_rest'      => true,
-				'sanitize_callback' => function ( $value ) {
-					$schema = array(
-						'type'    => 'string',
-						'enum'    => array( 'application', 'blog', 'user' ),
-						'default' => 'user',
-					);
-
-					if ( \is_wp_error( \rest_validate_enum( $value, $schema, '' ) ) ) {
-						return $schema['default'];
-					}
-
-					return $value;
-				},
-			)
+			self::get_activity_actor_meta_args()
```

Comment:
I understand why you did this, but it makes readability a lot worse! In that case I would prefer to do it the same way we did is with Extra_Fields: https://github.com/Automattic/wordpress-activitypub/blob/dbf0d11631fa74ba00b56c3ed51746d582cb40b6/includes/class-activitypub.php#L776